codec-http/src/main/java/io/netty/handler/codec/http/HttpHeaders.java

```diff
@@ -131,11 +154,32 @@ public HttpHeaders clear() {
         }
 
         @Override
+        public Entry<CharSequence, CharSequence> forEachEntry(EntryVisitor<CharSequence> visitor) throws Exception {
+            return null; // Since this is an empty header collection
+        }
+
+        @Override
         public Iterator<Entry<String, String>> iterator() {
             return entries().iterator();
         }
     };
 
+    private final EntryVisitor<CharSequence> setAllVisitor = new EntryVisitor<CharSequence>() {
```

Comment:
> Not that it matters in this case, but HttpHeaders isn't threadsafe, correct?

Correct. [DefaultHeaders](https://github.com/netty/netty/blob/master/codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java) is the base class for most header functionality and it is not thread safe. The [Headers](https://github.com/netty/netty/blob/master/codec/src/main/java/io/netty/handler/codec/Headers.java) currently does not guarantee thread safety.